lib/stats-collector.js

```diff
@@ -2,14 +2,22 @@
 
 /**
  * Provides a factory function for a {@link StatsCollector} object.
- * @private
  * @module
```

Comment:
```js
 * @module StatsCollector
```

Reply:
...`StatsCollector` is a typedef.  we don't want to conflicting names.  it seems like it should refer to what you'd actually call require with: `require('mocha/lib/stats-collector')`, so `stats-collector`.

it defaults to `lib/stats-collector`, which is fine for now, I guess.

Reply:
I mean, the docstrings need cleanup.  I am not convinced we're all on the same page about what belongs in the API docs and how best to present it.  ...or even how to come to that decision.  Maybe someone can just propose something and throw it in a PR and we can go from there?